qiskit/pulse/schedule.py

```diff
@@ -926,65 +478,1111 @@ def draw(self,
+        """Starting time of this schedule block."""
+        return self.ch_start_time(*self.channels)
+
+    @property
+    @_require_schedule_conversion
+    def stop_time(self) -> int:
+        """Stopping time of this schedule block."""
+        return self.duration
+
+    @property
+    def channels(self) -> Tuple[Channel]:
+        """Returns channels that this schedule clock uses."""
+        chans = set()
+        for block in self.instructions:
+            for chan in block.channels:
+                chans.add(chan)
+
+        return tuple(chans)
+
+    @property
+    def instructions(self) -> Tuple[BlockComponent]:
+        """Get the time-ordered instructions from self."""
+        return tuple(self._blocks)
+
+    @_require_schedule_conversion
+    def ch_duration(self, *channels: Channel) -> int:
+        """Return the time of the end of the last instruction over the supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_duration(*channels)
+
+    @_require_schedule_conversion
+    def ch_start_time(self, *channels: Channel) -> int:
+        """Return the time of the start of the first instruction over the supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_start_time(*channels)
+
+    @_require_schedule_conversion
+    def ch_stop_time(self, *channels: Channel) -> int:
+        """Return maximum start time over supplied channels.
+
+        Args:
+            *channels: Channels within ``self`` to include.
+        """
+        return self.ch_stop_time(*channels)
+
+    @_method_not_supported
+    def shift(self,
+              time: int,
+              name: Optional[str] = None,
+              inplace: bool = False):
+        """Return a schedule shifted forward by ``time``.
+
+        .. note:: This method is currently not supported.
+
+        Args:
+            time: Time to shift by.
+            name: Name of the new schedule. Defaults to the name of self.
+            inplace: Perform operation inplace on this schedule. Otherwise
+                return a new ``Schedule``.
+        """
+        pass
+
+    @_method_not_supported
+    def insert(self,
+               start_time: int,
+               schedule: ScheduleComponent,
+               name: Optional[str] = None,
+               inplace: bool = False):
+        """Return a new schedule with ``schedule`` inserted into ``self`` at ``start_time``.
+
+        .. note:: This method is currently not supported.
+
+        Args:
+            start_time: Time to insert the schedule.
+            schedule: Schedule to insert.
+            name: Name of the new schedule. Defaults to the name of self.
+            inplace: Perform operation inplace on this schedule. Otherwise
+                return a new ``Schedule``.
+        """
+        pass
+
+    def append(self, schedule: BlockComponent,
```

Comment:
```suggestion
    def append(self, block: BlockComponent,
```

Reply:
This violates signature of superclass. Do we disable linter?